whisper/src/lib.rs

```diff
@@ -586,50 +586,67 @@ fn file_fetch<F: Read + Seek>(fh: &mut F, header: &WhisperMetadata, archive: &Ar
     let interval = available.intersection(interval)
         .map_err(|s| io::Error::new(io::ErrorKind::Other, s))?;
 
-    __archive_fetch(fh, archive, interval.from(), interval.until()).map(Some)
+    let adjusted_interval = adjust_interval(&interval, archive.seconds_per_point)
+        .map_err(|s| io::Error::new(io::ErrorKind::Other, s))?;
+
+    __archive_fetch(fh, archive, adjusted_interval).map(Some)
+}
+
+fn adjust_instant(instant: u32, step: u32) -> u32 {
```

Comment:
It is not clear what this function should do.

Reply:
Looks like it is a discrete adjust to the left of the step

Reply:
it's just an extraction from __archive_fetch